### PR TITLE
Add MADV_POPULATE_{READ,WRITE}

### PR DESCRIPTION
**Pull request by Jesse-Bakker**

### CODE REVIEW

**Comment by adamreichold** [comment]
I think strictly speaking, our dependency on `libc` would need to be `0.2.143` instead of `0.2` since these two symbols are not present in `0.2.142`?

**Comment by Jesse-Bakker** [comment]
You are correct. I can change the dependency version, inline the constants, or close the PR, depending on if you want to depend on a newer version or not

File: src/advice.rs [hunk 1]
```diff
@@ -241,6 +241,71 @@ pub enum Advice {
     #[cfg(target_os = "linux")]
     HwPoison = libc::MADV_HWPOISON,
 
+    /// **MADV_POPULATE_READ** - Linux only (since Linux 5.14)
+    /// Populate  (prefault)  page  tables readable, faulting in all
```
Comment (RazrFalcon) [inline src/advice.rs:245]: Add new line.

File: src/advice.rs [hunk 2]
```diff
@@ -241,6 +241,71 @@ pub enum Advice {
...
+    /// permissions  and  special  mappings,  for  example,  mappings
+    /// marked with kernel-internal flags such as VM_PFNMAP or VM_IO,
+    /// or secret memory regions created using memfd_secret(2).
+    ///
+    /// Note  that with MADV_POPULATE_READ, the process can be killed
+    /// at any moment when the system runs out of memory.
+    #[cfg(target_os = "linux")]
+    PopulateRead = libc::MADV_POPULATE_READ,
+
+    /// **MADV_POPULATE_WRITE** - Linux only (since Linux 5.14)
```
Comment (RazrFalcon) [inline src/advice.rs:278]: Add new line.

**Comment by RazrFalcon** [comment]
Thanks.

**Pull request merged**